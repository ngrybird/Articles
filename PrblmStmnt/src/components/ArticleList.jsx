import React from "react";
import Article from "./Article";

class ArticleList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      order: "Ascending",
      articles: this.sortToAscending(this.props.articles)
    };
  }
  changeOrder = () => {
    if (this.state.order === "Ascending") {
      this.setState({
        order: "Descendnig",
        articles: this.sortToDescending(this.state.articles)
      });
    } else {
      this.setState({
        order: "Ascending",
        articles: this.sortToAscending(this.state.articles)
      });
    }
  };

  sortToAscending = dataList => {
    return dataList.sort((a, b) => {
      return new Date(a.date) - new Date(b.date);
    });
  };
  sortToDescending = dataList => {
    return dataList.sort((a, b) => {
      return new Date(b.date) - new Date(a.date);
    });
  };
  render() {
    return this.props.articles.length > 0 ? (
      <table>
        <tr>
          <th>Title</th>
          <th>Upvotes</th>
          <th>
            Release Date{" "}
            <button onClick={this.changeOrder}>
              {this.state.order === "Ascending" ? "Newest" : "Older"}
            </button>
          </th>
        </tr>
        {this.state.articles.map(item => (
          <Article key={item.articleId} articleData={item} />
        ))}
      </table>
    ) : (
      <div>No data found</div>
    );
  }
}
export default ArticleList;
