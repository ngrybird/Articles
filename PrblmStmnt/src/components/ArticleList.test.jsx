import React from "react";
import ArticleList from "./ArticleList";
import { fireEvent, render } from "@testing-library/react";
import { articles } from "../utils/Data";

describe("<ArticleList/>", () => {
  it("render Article component with sample data", () => {
    const { container, getByText } = render(
      <ArticleList articles={articles} />
    );

    expect(
      getByText("Harry Potter and the Sorcererâ€™s Stone Review")
    ).toBeInTheDocument();

    expect(getByText("56")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
