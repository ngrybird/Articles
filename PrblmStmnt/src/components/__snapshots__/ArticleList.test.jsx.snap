// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article/> render Article component with sample data 1`] = `
<div>
  <table>
    <tr>
      <th>
        Title
      </th>
      <th>
        Upvotes
      </th>
      <th>
        Release Date
         
        <button>
          Newest
        </button>
      </th>
    </tr>
    <tr>
      <td
        class="article-item"
      >
        Harry Potter and the Half Blood Prince Review
      </td>
      <td
        class="article-item"
      >
        23
      </td>
      <td
        class="article-item"
      >
        2016-12-01T18:30:00.000Z
      </td>
    </tr>
    <tr>
      <td
        class="article-item"
      >
        Harry Potter and the Sorcerer’s Stone Review
      </td>
      <td
        class="article-item"
      >
        56
      </td>
      <td
        class="article-item"
      >
        2016-12-22T18:30:00.000Z
      </td>
    </tr>
    <tr>
      <td
        class="article-item"
      >
        Harry Potter and the Goblet of Fire Review
      </td>
      <td
        class="article-item"
      >
        3
      </td>
      <td
        class="article-item"
      >
        2017-11-01T18:30:00.000Z
      </td>
    </tr>
  </table>
</div>
`;

exports[`<ArticleList/> render Article component with sample data 1`] = `
<div>
  <table>
    <tr>
      <th>
        Title
      </th>
      <th>
        Upvotes
      </th>
      <th>
        Release Date
         
        <button>
          Newest
        </button>
      </th>
    </tr>
    <tr>
      <td
        class="article-item"
      >
        Harry Potter and the Half Blood Prince Review
      </td>
      <td
        class="article-item"
      >
        23
      </td>
      <td
        class="article-item"
      >
        2016-12-01T18:30:00.000Z
      </td>
    </tr>
    <tr>
      <td
        class="article-item"
      >
        Harry Potter and the Sorcerer’s Stone Review
      </td>
      <td
        class="article-item"
      >
        56
      </td>
      <td
        class="article-item"
      >
        2016-12-22T18:30:00.000Z
      </td>
    </tr>
    <tr>
      <td
        class="article-item"
      >
        Harry Potter and the Goblet of Fire Review
      </td>
      <td
        class="article-item"
      >
        3
      </td>
      <td
        class="article-item"
      >
        2017-11-01T18:30:00.000Z
      </td>
    </tr>
  </table>
</div>
`;
